QLogger.cpp

```diff
@@ -1,266 +1,264 @@
+{
+    QDir dir(fileFolderDestination + QStringLiteral("/logs"));
+    if (!dir.exists()) {
+        return;
+    }
+
+    dir.setFilter(QDir::Files | QDir::Hidden | QDir::NoSymLinks);
+    const QFileInfoList list = dir.entryInfoList();
+
+    const QDateTime now = QDateTime::currentDateTime();
+
+    QFileInfoList::const_iterator i;
+    for (i = list.constBegin(); i != list.constEnd(); ++i) {
+        const QFileInfo fileInfo = *i;
+
+        if (fileInfo.lastModified().daysTo(now) >= days) {
+            //remove file
+            dir.remove(fileInfo.fileName());
+        }
+    }
+}
+
+void QLoggerManager::setDefaultFileDestinationFolder(const QString &fileDestinationFolder)
+{
+    mDefaultFileDestinationFolder = QDir::fromNativeSeparators(fileDestinationFolder);
 }
 
 void QLoggerManager::writeAndDequeueMessages(const QString &module)
 {
-   auto element = mNonWriterQueue.find(module);
-   const auto queueEnd = mNonWriterQueue.end();
-   const auto logWriter = getLogWriter(module);
+   QMutexLocker lock(&mMutex);
 
-   if (element != queueEnd && logWriter && !logWriter->isStop())
-   {
-      const auto module = element.key();
+   const auto logWriter = mModuleDest.value(module, Q_NULLPTR);
 
-      for (; element != queueEnd; ++element)
+   if (logWriter && !logWriter->isStop())
+   {
+      const auto values = mNonWriterQueue.values(module);
+      for (const auto &values : values)
       {
-         const auto message = element.value().at(0).toString();
-         const auto level = static_cast<LogLevel>(element.value().at(1).toInt());
-         const auto dt = element.value().at(2).toString();
+         const auto level = qvariant_cast<LogLevel>(values.at(2).toInt());
 
          if (logWriter->getLevel() <= level)
-            logWriter->write(module, message, level, dt);
+         {
+            const auto datetime = values.at(0).toDateTime();
+            const auto threadId = values.at(1).toString();
+            const auto function = values.at(3).toString();
+            const auto file = values.at(4).toString();
+            const auto line = values.at(5).toInt();
+            const auto message = values.at(6).toString();
+
+            logWriter->enqueue(datetime, threadId, module, level, function, file, line, message);
+         }
       }
 
       mNonWriterQueue.remove(module);
    }
 }
 
-void QLoggerManager::closeLogger()
+void QLoggerManager::enqueueMessage(const QString &module, LogLevel level, const QString &message, const QString& function,
+                                    const QString& file, int line)
 {
-   exit(0);
+   QMutexLocker lock(&mMutex);
+   const auto threadId = QString("%1").arg((quintptr)QThread::currentThread(), QT_POINTER_SIZE * 2, 16, QChar('0'));
+   const auto fileName = file.mid(file.lastIndexOf('/') + 1);
+   const auto logWriter = mModuleDest.value(module, Q_NULLPTR);
+
+   if (logWriter && logWriter->getMode() != LogMode::Disabled && !logWriter->isStop() && logWriter->getLevel() <= level)
```

Comment:
Now that I see it I would probably create a variable to wrapp some of the conditions like:
`const auto isLogEnabled = logWriter && logWriter->getMode() != LogMode::Disabled && !logWriter->isStop();`

and then:
`if (isLogEnabled && logWriter->getLevel() <= level)`

Reply:
Yes it is clearer